### PR TITLE
setitem no return value

### PR DESCRIPTION
**Pull request by geohotstan**
don't think setitem returns anything ([jax reference](https://github.com/google/jax/blob/main/jax/_src/basearray.pyi#L51))
a little unsure if this affects any previous DISK tensor implementations

also before the assert for contiguous was always returning True while not actually doing anything.
might've been the problem Cameron on discord was running into (I also accidentally ran into this thinking it was scheduler problem)

### CODE REVIEW

File: tinygrad/tensor.py [hunk 1]
```diff
@@ -808,15 +808,15 @@ def calc_dim(tensor_dim:int) -> int:
...
-  def __setitem__(self, indices, v:Union[Tensor, ConstType]):
-    if isinstance(self.device, str) and self.device.startswith("DISK"): return self.__getitem__(indices).assign(v)
+  def __setitem__(self, indices, v:Union[Tensor, ConstType]) -> None:
+    if isinstance(self.device, str) and self.device.startswith("DISK"): self.__getitem__(indices).assign(v)
     # TODO: broadcast v to the shape here, refactor for const v and one way broadcast_shape
     if not isinstance(v, Tensor): v = Tensor(v, self.device, self.dtype)
     assign_to = self.realize().__getitem__(indices)
     # NOTE: we check that indices is valid first
-    assert self.lazydata.contiguous(), "setitem target needs to be contiguous"
+    assert cast(LazyBuffer, self.lazydata).st.contiguous, "setitem target needs to be contiguous"
```
Comment (chenyuxyz) [inline tinygrad/tensor.py:817]: how does this work for multi? should it be `all(lb.st.contiguous for lb in self.lbs)`?

Comment (geohotstan) [reply tinygrad/tensor.py]: yeah that makes sense.

the other way is to do `self.lazydata = self.lazydata.contiguous()` to force contiguous inside setitem.
positive: don't need to `.contiguous()` in external implementation before setitem
negative: still need to `.contiguous()` before when using TinyJIT (for reasons I'm still trying to figure out) so there's a difference in usage...

File: tinygrad/tensor.py [hunk 2]
```diff
@@ -808,15 +808,16 @@ def calc_dim(tensor_dim:int) -> int:
         ret = ret.permute(*range(first_dim, first_dim+len(big_shape)), *range(0, first_dim), *range(first_dim+len(big_shape), ret.ndim))
     return ret
 
-  def __setitem__(self, indices, v:Union[Tensor, ConstType]):
-    if isinstance(self.device, str) and self.device.startswith("DISK"): return self.__getitem__(indices).assign(v)
+  def __setitem__(self, indices, v:Union[Tensor, ConstType]) -> None:
+    assert self.lazydata.st.contiguous if isinstance(self.lazydata, LazyBuffer) else all(lb.st.contiguous for lb in self.lazydata.lbs), \
```
Comment (chenyuxyz) [inline tinygrad/tensor.py:812]: `.lbs` works for single LazyBuffer too and make this line simpler

File: tinygrad/tensor.py [hunk 3]
```diff
@@ -808,15 +808,16 @@ def calc_dim(tensor_dim:int) -> int:
...
+      "setitem target needs to be contiguous"
+    if isinstance(self.device, str) and self.device.startswith("DISK"): self.__getitem__(indices).assign(v)
     # TODO: broadcast v to the shape here, refactor for const v and one way broadcast_shape
     if not isinstance(v, Tensor): v = Tensor(v, self.device, self.dtype)
     assign_to = self.realize().__getitem__(indices)
     # NOTE: we check that indices is valid first
```
Comment (chenyuxyz) [inline tinygrad/tensor.py:818]: update the NOTE too if you want to switch the order

File: tinygrad/tensor.py [hunk 4]
```diff
@@ -808,15 +808,18 @@ def calc_dim(tensor_dim:int) -> int:
...
     return ret
 
-  def __setitem__(self, indices, v:Union[Tensor, ConstType]):
-    if isinstance(self.device, str) and self.device.startswith("DISK"): return self.__getitem__(indices).assign(v)
-    # TODO: broadcast v to the shape here, refactor for const v and one way broadcast_shape
-    if not isinstance(v, Tensor): v = Tensor(v, self.device, self.dtype)
+  def __setitem__(self, indices, v:Union[Tensor, ConstType]) -> None:
+    if isinstance(self.device, str) and self.device.startswith("DISK"):
+      self.__getitem__(indices).assign(v)
+      return
```
Comment (geohotstan) [inline tinygrad/tensor.py:814]: maybe we just return the Tensor 🤣

**Comment by chenyuxyz** [comment]
thanks!

**Pull request merged**